src/components/account/details/details.js

```diff
@@ -8,151 +8,173 @@ import {
+      ) {
+        errors.email = "Invalid email address";
+      }
+
+      return errors;
+    }}
+    onSubmit={async (values, { setSubmitting, setUser }) => {
+      const client = createAuthenticatedClient();
+
+      setSubmitting(true);
+
+      const updatedUser = ({ email, createdAt, updatedAt, creator, ...rest }) =>
+        rest;
+
+      try {
+        const response = await client.request(updateUserById, {
+          params: updatedUser(values)
+        });
+
+        setUser({ ...response.updateUser });
+      } catch (error) {
+        console.log(error);
+      }
+    }}
+  >
+    {props => (
+      <Grid container spacing={1}>
+        <Grid item md={6} xs={12}>
+          <Field
+            name="email"
+            type="email"
+            label="Email"
+            variant="outlined"
+            margin="normal"
+            fullWidth
+            component={TextField}
+          />
+        </Grid>
+
+        <Grid item md={6} xs={12}>
+          <Field
+            disabled
+            name="password"
+            label="Password"
+            type="password"
+            variant="outlined"
+            margin="normal"
+            fullWidth
+            component={TextField}
+          />
+        </Grid>
+
+        <Grid item md={6} xs={12}>
+          <Field
+            name="firstName"
+            label="First Name"
+            variant="outlined"
+            margin="normal"
+            fullWidth
+            component={TextField}
+          />
+        </Grid>
+
+        <Grid item md={6} xs={12}>
+          <Field
+            name="secondName"
+            label="Second Name"
+            variant="outlined"
+            margin="normal"
+            fullWidth
+            component={TextField}
+          />
+        </Grid>
+
+        <Grid item md={6} xs={12}>
+          <Field
+            name="matriculation"
+            label="Matriculation"
+            variant="outlined"
+            margin="normal"
+            fullWidth
+            component={TextField}
+          />
+        </Grid>
+
+        <Grid item md={6} xs={12}>
+          <Field
+            name="cpf"
+            label="CPF"
+            variant="outlined"
+            margin="normal"
+            fullWidth
+            component={TextField}
+          />
+        </Grid>
+
+        {props.isSubmitting && <LinearProgress />}
+
+        <br />
+
+        <Divider />
+
+        <CardActions>
+          <Button
+            type="submit"
+            variant="contained"
+            color="primary"
+            fullWidth
+            className={classes.submit}
+            disabled={props.isSubmitting}
+            onClick={props.submitForm}
+          >
+            Update
+          </Button>
+        </CardActions>
+      </Grid>
+    )}
+  </Formik>
+);
+
 const AccountDetails = props => {
   const { className, ...rest } = props;
 
   const classes = useStyles();
 
-  const [values, setValues] = useState({
-    firstName: "Shen",
```

Comment:
I'm sorry, avoid what?
You mean `const classes = useStyles();` ?

Reply:
> Try to avoid this in the future, because later on we need to go back and remove or we forget to do so

@Sphinxs  I'm sorry, avoid what?
You mean `const classes = useStyles();` ?

Reply:
Values like theses ones:

```
lastName: "Zhi",	
email: "shen.zhi@devias.io",	
phone: "",	
state: "Alabama",	
country: "USA"
```

The so caled 'mock' ( :

Reply:
> Values like theses ones:
> 
> ```
> lastName: "Zhi",	
> email: "shen.zhi@devias.io",	
> phone: "",	
> state: "Alabama",	
> country: "USA"
> ```
> 
> The so caled 'mock' ( :

This was taken from a template.
I just forgot to delete it.